### DISCUSSION TITLE
New Feature: below Wh - Switch off

### DISCUSSION THREAD

**Discussion by Rappelsack** [mongoose-os-apps/shelly-homekit#1470] [Ideas]
Hello

I do not know how to describe it correctly.
It would e a awesome feature it f.e. a Shelly Plug S could be set up to: below XY Wh, cut the current.

So I could plug my TV, activate the Shelly and when I set the TV to standby the current drops below a limit and after that the shelly switches off.

Hope u understand.

Kind regards.

**Comment by andyblac** [comment]
plesse add your support to this issue https://github.com/mongoose-os-apps/shelly-homekit/issues/26, i'll close this one as duplicate thx.

**Comment by Rappelsack** [comment]
Thanks!

I do not mean the power consumption readout in homekit.

I mean a simmilar function like „auto off delay“:
![D9A734E8-0406-43BF-9B5D-DB5C0B66A808](https://user-images.githubusercontent.com/67721205/106355486-90761000-62f8-11eb-963a-457531a98bc0.jpeg)

But „auto off if the consumption drops below 0,5W“ for example.

**Comment by timoschilling** [comment]
@rojer what do you think about this issue? I think thats something which should be handelt by homekit after implementing #26

**Comment by Rappelsack** [comment]
Any news about this?

I guess it should be not difficult to implement?

**Comment by rojer** [comment]
well, homekit does not provide any means of exporting consumption, let alone controlling it...
there are no fundamental objections, but we need to persist consumption across reboots.
other than that, it would indeed be pretty easy to do.

**Comment by andyblac** [comment]
> well, homekit does not provide any means of exporting consumption, let alone controlling it...

not exactly true, Home app desn't, but homekit does, via 3rd party app's (Eve, Home+, etc)

**Comment by Rappelsack** [comment]
Well, it doesn't have to go over HomeKit. It's via the web GUI. Just like the auto-Off. 

That's how you could connect your TV. If the power consumption falls below XY watts, because you set it into standby via remote control the Shelly turns off. It could be switched on again with a HomeKit button.

**Comment by timoschilling** [comment]
@rojer are you still think we should implement this? After #690 this can be done by and HomeKit automatisation.

**Comment by rojer** [comment]
but that's for eve only, isn't it?
i think it's still a good idea to have locally, since it's a safety mechanism as well.

**Comment by timoschilling** [comment]
Not only for eve, there are many apps that can create automatisations for that.
Why is below power cutoff a safety mechanism? Keep in mind over power is requested in a different issue #339

**Comment by rojer** [comment]
oh, sorry. indeed, i got confused, i thought this was for overpower protection.
but i guess if we are to implement the overpower cutoff, might as well add a check for below certain threshold.

**Comment by Rappelsack** [comment]
I see that the development is a little bit behind? But is there any news about this function?
It would be so great to cut the current via Shelly S when the washing machine is ready (consumption below XY Wh) f.e.
